circle.yml

```diff
@@ -151,6 +151,49 @@ commands:
           name: Restore all node_modules to proper workspace folders
           command: node scripts/circle-cache.js --action unpack
 
+  restore_cached_system_tests_deps:
+    steps:
+      - run:
+          name: Generate Circle Cache key for system tests
+          command: yarn workspace @tooling/system-tests cache:key > system_tests_cache_key
+      - restore_cache:
+          name: Restore system tests node_modules cache
+          keys:
+            - v{{ .Environment.CACHE_VERSION }}-{{ arch }}-system-tests-node-modules-cache-{{ checksum "system_tests_cache_key" }}
+            - v{{ .Environment.CACHE_VERSION }}-{{ arch }}-system-tests-node-modules-cache-
```

Comment:
https://circleci.com/docs/2.0/caching/#restoring-cache

> CircleCI restores caches in the order of keys listed in the restore_cache step. Each cache key is namespaced to the project, and retrieval is prefix-matched. The cache will be restored from the first matching key. If there are multiple matches, the most recently generated cache will be used.

So it will either restore the exact correct cache, or the latest created cache.